plugins/woocommerce/src/Internal/Admin/EmailPreview/EmailPreviewRestController.php

```diff
@@ -98,6 +97,49 @@ public function register_routes() {
 				),
 			)
 		);
+
+		register_rest_route(
+			$this->route_namespace,
+			'/' . $this->rest_base . '/save-transient',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => fn( $request ) => $this->save_transient( $request ),
+					'permission_callback' => fn( $request ) => $this->check_permissions( $request ),
+					'args'                => array(
+						'key'   => array(
+							'required'          => true,
+							'type'              => 'string',
+							'description'       => 'The key for the transient. Must be one of the allowed options.',
+							'validate_callback' => function ( $key ) {
+								if ( ! in_array( $key, EmailPreview::EMAIL_SETTINGS_IDS, true ) ) {
+									return new \WP_Error(
+										'woocommerce_rest_not_allowed_key',
+										sprintf( 'The provided key "%s" is not allowed.', $key ),
+										array( 'status' => 400 ),
+									);
+								}
+								return true;
+							},
+							'sanitize_callback' => 'sanitize_text_field',
+						),
+						'value' => array(
+							'required'          => true,
+							'type'              => 'string',
+							'description'       => 'The value to be saved for the transient.',
+							'validate_callback' => 'rest_validate_request_arg',
+							'sanitize_callback' => function ( $value, $request ) {
+								$key = $request->get_param( 'key' );
+								if ( 'woocommerce_email_footer_text' === $key ) {
+									return sanitize_textarea_field( $value );
```

Comment:
If I have a `<br />` tag in my footer text, it works fine on the first page load, but disappears (no line break, two lines become one) if I do any changes to footer text. It looks like the tag gets lost when saving the transient.

Reply:
Fixed in 808996f. There were different sanitization functions when saving option and transient.